### PR TITLE
KAFKA-17585: `offsetResetStrategyTimestamp` should return `long` instead of `Long`

### PR DESCRIPTION
**Pull request by xijiu**
As title

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcherUtils.java [hunk 1]
```diff
@@ -234,9 +234,8 @@ Map<TopicPartition, Long> getOffsetResetTimestamp() {
         Set<TopicPartition> partitions = subscriptionState.partitionsNeedingReset(time.milliseconds());
         final Map<TopicPartition, Long> offsetResetTimestamps = new HashMap<>();
         for (final TopicPartition partition : partitions) {
-            Long timestamp = offsetResetStrategyTimestamp(partition);
-            if (timestamp != null)
-                offsetResetTimestamps.put(partition, timestamp);
+            long timestamp = offsetResetStrategyTimestamp(partition);
+            offsetResetTimestamps.put(partition, timestamp);
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcherUtils.java]: nit: could you please remove the local variable `timestamp`?
```java
offsetResetTimestamps.put(partition, offsetResetStrategyTimestamp(partition));
```

**Review by chia7712 — COMMENTED** [review]
@xijiu thanks for this patch

**Comment by xijiu** [comment]
@chia7712  Sure, I have removed it,   PTAL

**Review by chia7712 — COMMENTED** [review]
LGTM

**Pull request merged**